styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -512,28 +514,29 @@ private void setupMetrics(
       RateLimiter submissionRateLimiter,
       Stats stats) {
 
-    final Gauge<Long> queuedEventsCount = stateManager::getQueuedEventsCount;
-    final Gauge<Long> allWorkflowsCount = () -> workflowCache.all().stream().count();
-    final Gauge<Long> configuredWorkflowsCount = () -> workflowCache.all().stream()
+    stats.registerQueuedEvents(stateManager::getQueuedEventsCount);
+
+    stats.registerWorkflowCount("all", () -> (long) workflowCache.all().size());
+
+    stats.registerWorkflowCount("configured", () -> workflowCache.all().stream()
         .filter(WorkflowValidator::hasDockerConfiguration)
-        .count();
-    final Gauge<Long> configuredEnabledWorkflowsCount = () -> {
-      try {
-        final Set<WorkflowId> enabledWorkflowsSet = storage.enabled();
-        return workflowCache.all().stream()
-            .filter(WorkflowValidator::hasDockerConfiguration)
-            .filter((workflow) -> enabledWorkflowsSet.contains(WorkflowId.ofWorkflow(workflow)))
-            .count();
-      } catch (IOException e) {
-        LOG.error("Failed to read enabled status from storage", e);
-        return 0L;
-      }
+        .count());
+
+    final Supplier<Gauge<Long>> configuredEnabledWorkflowsCountGaugeSupplier = () -> {
+      final Supplier<Set<WorkflowId>> enabledWorkflowSupplier =
+          new CachedSupplier<>(storage::enabled, Instant::now);
```

Comment:
Oh, I didn't know we already had `CachedSupplier`. Much better. And it logs the failures, which allows some log based alerting...